### PR TITLE
Windows fixes

### PR DESCRIPTION
**Pull request by ArcaneNibble**

### CODE REVIEW

File: log.h [hunk 1]
```diff
@@ -30,6 +30,11 @@
 #include <string>
 #include <vector>
 #include <mutex>
+#ifdef _WIN32
+// Include mingw-std-threads extra header
+#include <mingw.mutex.h>
```
Comment (whitequark) [inline log.h]: why is this needed?

Comment (ArcaneNibble) [reply log.h]: So mingw has two different threading "ABIs" for Windows: posix and native. The "posix" mode works by having a library that emulates the pthreads API using the Windows native threading capability whereas the "native" mode just doesn't have that. However, libstdc++ implements the C++11 threading interfaces using the pthreads API, so the "native" threading mode version of MinGW just doesn't have the C++ threads API available at all. So you now have two choices: a) use the "posix" version of the mingw compiler (and become dependent on the pthreads-win32 code) or b) use the "native" version of the mingw compiler (and become dependent on some other implementation of C++11 threads (which is mingw-std-threads in this case)). I somewhat-arbitrarily decided I liked option (b) better (despite the fact that yosys-abc still ends up requiring pthreads-win32 anyways).

Comment (whitequark) [reply log.h]: Why is this gated on _WIN32 and not __MINGW32__ though? This would break, e.g., an MSVC build, or a Clang build.

**Comment by ArcaneNibble** [comment]
Fixed to only include mingw-std-threads when *) actually compiling with mingw and *) not using mingw posix threads.
